### PR TITLE
Feature/rework postgres block query with shared model

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
### Description of the Change
migrates postgres_block_query to the shared_model

### Benefits
more shared_model

### Possible Drawbacks 
no

### Usage Examples or Tests *[optional]*
no

### Alternate Designs *[optional]*
no

### CODE REVIEW

**Comment by x3medima17** [comment]
You can remove unused optional paragraphs from the PR description

File: irohad/ametsuchi/impl/postgres_block_query.cpp [hunk 1]
```diff
@@ -39,24 +42,23 @@ namespace iroha {
...
-              s.on_completed();
-            });
+        if (not block) {
+          log_->error("error while converting from JSON");
+          // TODO load corrupted block from ledger
+        }
+
+        return rxcpp::observable<>::create<PostgresBlockQuery::wBlock>([block{
+            std::move(block)}](auto s) {
+          if (block)
```
Comment (x3medima17) [inline irohad/ametsuchi/impl/postgres_block_query.cpp]: Please use {} even if there is only one line

File: irohad/ametsuchi/impl/storage_impl.cpp [hunk 2]
```diff
@@ -271,13 +271,18 @@ DROP TABLE IF EXISTS index_by_id_height_asset;
       auto storage_ptr = std::move(mutableStorage);  // get ownership of storage
       auto storage = static_cast<MutableStorageImpl *>(storage_ptr.get());
       for (const auto &block : storage->block_store_) {
-        // TODO: rework to shared model converters once they are available
-        // IR-1084 Nikita Alekseev
-        auto old_block =
-            *std::unique_ptr<model::Block>(block.second->makeOldModel());
+        std::string ss = shared_model::converters::protobuf::modelToJson(
+            *std::static_pointer_cast<shared_model::proto::Block>(
```
Comment (x3medima17) [inline irohad/ametsuchi/impl/storage_impl.cpp]: Maybe it is better to use dynamic pointer cast?

File: irohad/ametsuchi/impl/postgres_block_query.cpp [hunk 3]
```diff
@@ -112,13 +114,15 @@ namespace iroha {
...
-          // model
-          return boost::optional<shared_model::proto::Block>(
-              shared_model::proto::from_old(*serializer_.deserialize(
-                  *model::converters::stringToJson(bytesToString(bytes)))));
+          return shared_model::converters::protobuf::jsonToModel<
+              shared_model::proto::Block>(bytesToString(bytes));
         };
+        if (not block) {
+          log_->error("error while converting from JSON");
+          // TODO load corrupted block from ledger
```
Comment (nickaleks) [inline irohad/ametsuchi/impl/postgres_block_query.cpp]: Add task to JIRA pls

File: irohad/ametsuchi/impl/storage_impl.cpp [hunk 4]
```diff
@@ -271,13 +271,15 @@ DROP TABLE IF EXISTS index_by_id_height_asset;
...
       auto storage = static_cast<MutableStorageImpl *>(storage_ptr.get());
       for (const auto &block : storage->block_store_) {
-        // TODO: rework to shared model converters once they are available
-        // IR-1084 Nikita Alekseev
-        auto old_block =
-            *std::unique_ptr<model::Block>(block.second->makeOldModel());
-        block_store_->add(block.first,
-                          stringToBytes(model::converters::jsonToString(
-                              serializer_.serialize(old_block))));
+        std::string ss = shared_model::converters::protobuf::modelToJson(
```
Comment (nickaleks) [inline irohad/ametsuchi/impl/storage_impl.cpp]: Why we need SS here? Maybe use it in add call?

File: test/module/irohad/ametsuchi/ametsuchi_test.cpp [hunk 5]
```diff
@@ -727,13 +727,14 @@ TEST_F(AmetsuchiTest, FindTxByHashTest) {
...
   auto tx3hash = shared_model::crypto::Hash("some garbage");
 
-  auto tx1check = *blocks->getTxByHashSync(tx1hash);
+  auto tx1 = blocks->getTxByHashSync(tx1hash);
+  ASSERT_TRUE(tx1);
 
-  auto tx1 = *blocks->getTxByHashSync(tx1hash);
-  auto tx2 = *blocks->getTxByHashSync(tx2hash);
+  auto tx2 = blocks->getTxByHashSync(tx2hash);
+  ASSERT_TRUE(tx1);
```
Comment (nickaleks) [inline test/module/irohad/ametsuchi/ametsuchi_test.cpp]: tx2 here?

File: irohad/ametsuchi/impl/postgres_block_query.cpp [hunk 6]
```diff
@@ -196,30 +198,28 @@ namespace iroha {
...
-          }
-      | [&](const auto &json) { return serializer_.deserialize(json); } |
-          [](const auto &block) {
-            return boost::optional<shared_model::proto::Block>(
-                shared_model::proto::from_old(block));
-          }
-      | [&](const auto &block) {
+      auto block = getBlockId(hash) |
+          [this](auto block_id) { return block_store_.get(block_id);
+         } | [this](auto bytes) {
```
Comment (nickaleks) [inline irohad/ametsuchi/impl/postgres_block_query.cpp]: auto defaults to pass-by-value, consider using auto &bytes. Also check other lambdas

**Pull request merged**